src/its_preselector/preselector.py

```diff
@@ -77,62 +79,114 @@ def __set_cal_sources(self, cal_sources):
             cal_source = CalSource(c)
             self.cal_sources.append(cal_source)
 
-    def get_frequency_low_passband(self, rf_path_index):
-        if rf_path_index < len(self.rf_paths):
-            path = self.rf_paths[rf_path_index]
+    def get_frequency_low_passband(self, rf_path_name: str) -> float:
+        """
+         Get the low frequency of the 1 dB passband.
+        :param rf_path_name, or name of the rf_path:
+        :return:The low frequency of hte 1dB passband in Hz.
+        """
+        if rf_path_name in self.rf_paths:
+            path = self.rf_paths[rf_path_name]
             filter_id = path.filter_id
-            preselctor_filter = self.__get_filter(filter_id)
-            if preselctor_filter:
-                return preselctor_filter.frequency_low_passband
-        return None
-
-    def get_frequency_high_passband(self, rf_path_index):
-        if rf_path_index < len(self.rf_paths):
-            path = self.rf_paths[rf_path_index]
+            preselector_filter = self.__get_filter(filter_id)
+            if preselector_filter:
+                return preselector_filter.frequency_low_passband
+            else:
```

Comment:
This `else` should be un-indented one level, so it corresponds to the `if rf_path_name in self.rf_paths` statement on line 88. 

Another `else` should probably be added to handle the situation where the RF path is present but there is an error getting the filter.

Reply:
Fixed the first thing I mentioned in [0377439](https://github.com/NTIA/Preselector/pull/11/commits/03774397354fc9ca6366560685964e4b9a75859d)

I'll leave the rest up to you, since there are multiple places where similar handling may need to be added (see other comments)

Reply:
Should be addressed in https://github.com/NTIA/Preselector/commit/56430e268ca526366e095d17797655a65ceb1ba8 and https://github.com/NTIA/Preselector/commit/539a5b5a6b7f75452380d649284539fac55d7eaf